### PR TITLE
only raise piggyback count if ping is succesfull

### PR DESCRIPTION
**Pull request by CorgiMan**
Not ready for review yet, only want to see if it passes the tests

### CODE REVIEW

**Comment by CorgiMan** [comment]
ready for review @uber/ringpop

File: package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "Jake Verbaten <raynos2@gmail.com>",
     "Mark Yen <mark@uber.com>"
   ],
-  "version": "10.13.1",
+  "version": "10.14.0",
```
Comment (jwolski) [inline package.json]: Don't bump until merged.

File: lib/membership/index.js [hunk 2]
```diff
@@ -304,7 +305,11 @@ Membership.prototype.update = function update(changes, isLocal) {
 
             // localMember is carried around as a convenience.
             if (member.address === this.ringpop.whoami()) {
+                // If localMember is not yet defined, source and
+                // sourceIncarnationNumber have not yet been set.
                 this.localMember = member;
+                change.source = member.address;
```
Comment (jwolski) [inline lib/membership/index.js]: This is very dirty. Why is this required?

File: lib/gossip/dissemination.js [hunk 3]
```diff
@@ -81,12 +81,34 @@ Dissemination.prototype.isEmpty = function isEmpty() {
...
+                local: self.ringpop.whoami(),
+                err: err,
+                numMembershipChanges: membershipChanges.length
+            });
+            return;
+        }
+
+        for (var i = 0; i < membershipChanges.length; i++) {
+            var change = membershipChanges[i];
+            if (self.changes[change.address] && self.changes[change.address].id === change.id) {
```
Comment (jwolski) [inline lib/gossip/dissemination.js]: you can DRY this up with:

```
var issuedChange = membershipChanges[i];
var localChange = self.changes[issuedChange.address];
if (localChange && localChange.id === issuedChange.id) {
  localChange.piggybackCount++
}
```

File: lib/gossip/dissemination.js [hunk 4]
```diff
@@ -130,6 +152,7 @@ Dissemination.prototype.onRingChanged = function onRingChanged() {
 
 Dissemination.prototype.recordChange = function recordChange(change) {
     this.changes[change.address] = change;
+    this.changes[change.address].piggybackCount = 0;
```
Comment (jwolski) [inline lib/gossip/dissemination.js]: This modifies the caller-provided change... which is not great from an API perspective.

File: lib/gossip/dissemination.js [hunk 5]
```diff
@@ -81,12 +81,34 @@ Dissemination.prototype.isEmpty = function isEmpty() {
...
-    }
+    });
+    
+    issue(membershipChanges, function onIssue(err) {    
+        // Raise the piggyback count if the local change has been sent
+        // successfully. When the local change has been updated we don't
+        // want to raise the piggyback count. We can deduce if the change
+        // has been updated by comparing the id's of the changes.
+        if (err) {
+            self.ringpop.logger.info('ringpop dissemination not bumping piggyback count', {
```
Comment (jwolski) [inline lib/gossip/dissemination.js]: Stat plz

File: lib/gossip/dissemination.js [hunk 6]
```diff
@@ -81,12 +81,36 @@ Dissemination.prototype.isEmpty = function isEmpty() {
...
+    });
+
+    issue(membershipChanges, function onIssue(err) {
+        // Raise the piggyback count if the local change has been sent
+        // successfully. When the local change has been updated we don't
+        // want to raise the piggyback count. We can deduce if the change
+        // has been updated by comparing the id's of the changes.
+        if (err) {
+            self.ringpop.stat('increment', 'dont-bump-piggyback-count');
+            self.ringpop.logger.info('ringpop dissemination not bumping piggyback count', {
```
Comment (jwolski) [inline lib/gossip/dissemination.js]: make this use a dissemination logger (not the default ringpop logger).
